core/modules/util/ConfigStore.h

```diff
@@ -76,31 +89,60 @@ class ConfigStore {
     friend std::ostream& operator<<(std::ostream &out, ConfigStore const& config);
 
     /** Get value for a configuration key
+     *
      * @param key configuration key
      * @return the string value for a key
-     * @throw a ConfigError exception if key is not found
+     * @throw KeyNotFoundError if key is not found
      */
-    std::string get(std::string const& key) const;
+    std::string getString(std::string const& key) const;
 
-    /** Get value for a configuration key
+    /** Get value for a configuration key or a default value if key is not found
+     *
      * @param key configuration key
      * @params defaultValue
      * @return the string value for a key, defaulting to defaultValue
      */
-    std::string get(std::string const& key, std::string const& defaultValue) const;
+    std::string getStringOrDefault(std::string const& key,
+        std::string const& defaultValue = std::string()) const;
 
-    /// @return the typed value for a key, defaulting to defaultValue
-    int getInt(std::string const& key, int const& defaultValue = 0) const;
+    /** Get value for a configuration key or a default value if key is not found
+     *
+     * @param key configuration key
+     * @params defaultValue
+     * @return the integer value for a key, defaulting to defaultValue
+     *
+     * @throw InvalidIntegerValue if value can not be converted to an integer
+     */
+    int getIntOrDefault(std::string const& key, int const& defaultValue = 0) const;
 
-    const lsst::qserv::StringMap& getConfigMap() const {
-        return _configMap;
+    /** Get a collection of (key, value) related to a configuration section
+     *
+     *  All ConfigStore entries having key like "section.param_key" are returned
+     *  but all key name are shortened to "param_key"
+     *
+     * @param sectionName name of configuration section
+     * @return a collection of (key, value) related to a configuration section
+     *
+     */
+    std::map<std::string, std::string> const getSectionConfigMap(std::string sectionName) const {
```

Comment:
`const` in return type does not make much sense. I'd move this method to .cc file, in-lining will not likely to happen for this method in any case.